From 1ba749bbf9a06766e35d632967d61ec22078e9f2 Mon Sep 17 00:00:00 2001
From: Mike Galbraith <mgalbraith@suse.de>
Date: Tue, 27 Sep 2016 09:42:47 +0200
Subject: [PATCH] kernel/zram: Fix testcase for kernels >= v4.7

max_comp_streams is depreciated in kernels >= v4.7, bail instead
of failing.

Signed-off-by: Mike Galbraith <mgalbraith@suse.de>
Acked-by: Cyril Hrubis <chrubis@suse.cz>
---
 testcases/kernel/device-drivers/zram/zram_lib.sh | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/testcases/kernel/device-drivers/zram/zram_lib.sh b/testcases/kernel/device-drivers/zram/zram_lib.sh
index d3e62e0..2144227 100755
--- a/testcases/kernel/device-drivers/zram/zram_lib.sh
+++ b/testcases/kernel/device-drivers/zram/zram_lib.sh
@@ -62,6 +62,14 @@ zram_load()
 
 zram_max_streams()
 {
+	tst_kvercmp 4 7 0
+	if [ $? -gt 0 ]; then
+		tst_resm TCONF "device attribute max_comp_streams is"\
+			"depriciated since kernel v4.7, the running kernel"\
+			"does not support it"
+		return
+	fi
+
 	tst_kvercmp 3 15 0
 	if [ $? -eq 0 ]; then
 		tst_resm TCONF "device attribute max_comp_streams is"\
-- 
1.8.3.1

